### ISSUE TITLE
Community Call

### ISSUE THREAD

**Issue by swetoast** [custom-components/sensor.rpi_power#15]
Hey Peeps since the forum thread is closed after Balloob not handling critique i thought it might be a good thing to have an open thread here about changes and what you guys want for the component.

My plans are to change up the error messages to make it more readable the common user since the values displayed doesn't help the common user and the error messages could be better so that it tells the novice user that their psu is good or bad but i want the community help for this so its not just me doing this and nobody wanting it.

**Comment by Beduir** [comment]
I always get one error message:
_There is a problem with your power supply or system._
The error codes are different:
- 50000
- 50005
- 80000
This is normal?
I was watching the code. It compares the error number with a 4-digit number.
`        elif _throttled == '5000':
            self._description = 'CPU is throttled due to under-voltage.'
        elif _throttled == '8000':
            self._description = 'Soft Temp limit has occurred.'`

**Comment by swetoast** [comment]
its not good, that means its time to replace your power supply 

50000 to 50005 means the power supply isnt providing the raspberry pi with enough power that can lead to corruption and crashing

80000 is overheating

**Comment by Beduir** [comment]
Thank you, I know about the power supply.
Now I'm talking about the text description.
For the code 50005, the description "There is a problem with your power supply or system." is displayed.
I think it should be "CPU is throttled due to under-voltage.".

**Comment by swetoast** [comment]
"The Raspberry Pi is throttled due to a bad power supply, please replace your changer"

How about that ?

**Comment by Beduir** [comment]
I guess I didn’t explain well. English is not native and I write through a translator.
The text descriptions suits me.

For any codes, only two descriptions are displayed:
0 = Working as intended
50005, 50000, 80000 = There is a problem with your power supply or system.
I expected to see:
0 = Working as intended
50005, 50000 = CPU is throttled due to under-voltage.
80000 = Soft Temp limit has occurred.

`_throttled = open(SYSFILE, 'r').read()[:-1]   # I have here _throttled == 50005 or 50000 or 80000`

`        # this code is always executed :
        else:
            self._description = 'There is a problem with your power supply or system.'
`

**Comment by swetoast** [comment]
Okey this thread is more or less about changes in the component not a support thread all you need to do atm is change your powersupply

please open another thread if you want support

**Comment by zombu2** [comment]
then get a new psu with more A and a passive heat sink

**Comment by aluisperezh** [comment]
> Okey this thread is more or less about changes in the component not a support thread all you need to do atm is change your powersupply
> 
> please open another thread if you want support

I opened another thread because I have the same problem in my system, clearly stating that I KNOW MY POWER SUPPLY IS FAULTY and that I have already ordered a new one. But the component is not working as it should, because it should display another message and not an error in logs. For me that means there is an issue with the code. Why has it been closed with "your powersupply is faulty"?
